pkg/scalers/couchdb_scaler.go

```diff
@@ -19,216 +18,173 @@ import (
 	}
-	metricSpec := v2.MetricSpec{
-		External: externalMetric, Type: externalMetricType,
+
+	meta, err := parseCouchDBMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing couchdb metadata: %w", err)
 	}
-	return []v2.MetricSpec{metricSpec}
-}
 
-func (s couchDBScaler) Close(ctx context.Context) error {
-	if s.client != nil {
-		err := s.client.Close(ctx)
-		if err != nil {
-			s.logger.Error(err, fmt.Sprintf("failed to close couchdb connection, because of %v", err))
-			return err
-		}
+	connStr := meta.ConnectionString
+	if connStr == "" {
+		addr := net.JoinHostPort(meta.Host, meta.Port)
+		connStr = "http://" + addr
 	}
-	return nil
-}
 
-func (s *couchDBScaler) getQueryResult(ctx context.Context) (int64, error) {
-	db := s.client.DB(ctx, s.metadata.dbName)
-	var request couchDBQueryRequest
-	err := json.Unmarshal([]byte(s.metadata.query), &request)
+	client, err := kivik.New("couch", connStr)
 	if err != nil {
-		s.logger.Error(err, fmt.Sprintf("Couldn't unmarshal query string because of %v", err))
-		return 0, err
+		return nil, fmt.Errorf("error creating couchdb client: %w", err)
 	}
-	rows, err := db.Find(ctx, request, nil)
+
+	err = client.Authenticate(ctx, couchdb.BasicAuth("admin", meta.Password))
 	if err != nil {
-		s.logger.Error(err, fmt.Sprintf("failed to fetch rows because of %v", err))
-		return 0, err
+		return nil, fmt.Errorf("error authenticating with couchdb: %w", err)
 	}
-	var count int64
-	for rows.Next() {
-		count++
-		res := Res{}
-		if err := rows.ScanDoc(&res); err != nil {
-			s.logger.Error(err, fmt.Sprintf("failed to scan the doc because of %v", err))
-			return 0, err
-		}
+
+	isConnected, err := client.Ping(ctx)
+	if !isConnected || err != nil {
+		return nil, fmt.Errorf("failed to ping couchdb: %w", err)
 	}
-	return count, nil
+
+	return &couchDBScaler{
+		metricType: metricType,
+		metadata:   meta,
+		client:     client,
+		logger:     InitializeLogger(config, "couchdb_scaler"),
+	}, nil
 }
 
-func parseCouchDBMetadata(config *scalersconfig.ScalerConfig) (*couchDBMetadata, string, error) {
-	var connStr string
-	var err error
+func parseCouchDBMetadata(config *scalersconfig.ScalerConfig) (couchDBMetadata, error) {
 	meta := couchDBMetadata{}
-
-	if val, ok := config.TriggerMetadata["query"]; ok {
-		meta.query = val
-	} else {
-		return nil, "", fmt.Errorf("no query given")
+	err := config.TypedConfig(&meta)
+	if err != nil {
+		return meta, fmt.Errorf("error parsing couchdb metadata: %w", err)
 	}
 
-	if val, ok := config.TriggerMetadata["queryValue"]; ok {
-		queryValue, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, "", fmt.Errorf("failed to convert %v to int, because of %w", val, err)
-		}
-		meta.queryValue = queryValue
-	} else {
-		if config.AsMetricSource {
-			meta.queryValue = 0
-		} else {
-			return nil, "", fmt.Errorf("no queryValue given")
-		}
+	err = meta.Validate()
+	if err != nil {
+		return meta, err
```

Comment:
shouldn't be necessary to call explicitly as it should be called implicitly as part of `TypedConfig()` invocation

https://github.com/kedacore/keda/blob/b2ce95daaa8d5f8b4842561c594b8767a7194bd4/pkg/scalers/scalersconfig/typed_config.go#L184-L188

Reply:
Removed, thanks for the feedback!

Reply:
Hi @wozniakjan . Can you re-rerun the `/run-e2e couchdb`?